src/System.Private.CoreLib/shared/System/Exception.cs

```diff
@@ -119,27 +120,52 @@ public virtual void GetObjectData(SerializationInfo info, StreamingContext conte
 
         public override string ToString()
         {
-            string s = GetClassName();
+            // Get the lengths of the StackTrace and _innerException first so we know how large to make the ValueStringBuilder
+            string? stackTrace = StackTrace;
+            string? innerException = null;
+            if (_innerException != null)
+            {
+                innerException = _innerException.ToString();
+            }
+
+            int initialLength = 128;
+            if (stackTrace != null)
+            {
+                initialLength += stackTrace.Length;
+            }
+            if (innerException != null)
+            {
+                initialLength += innerException.Length;
+            }
+
+            ValueStringBuilder sb = new ValueStringBuilder(initialLength);
```

Comment:
Just for my info, do we expect any efficiency gain from going from StringBuilderCache to using VSB without a stackalloc ? Or just cleanup/reduce duplicate implementation?

Reply:
Yes. StringBuilderCache works for up to 360 chars only. It has worse perf characteristics than just allocating StringBuilder every time for larger sizes.

The full string is often going to be more than 360 chars here.

Reply:
SBC is also problematic when it goes nested (say for formatting) as something has already taken the threadstatic copy.

Reply:
What about use of StringBuilder itself in corelib? Where code is not using the formatting features of StringBuilder, and not accepting it or passing it back, we could potentially replace with VSB and get the benefits of the array pool. (As a separate PR of course.)

Reply:
Yep!